src/deepsparse/transformers/utils/helpers.py

```diff
@@ -211,3 +212,91 @@ def create_causal_mask(
     causal_mask[:, :, :, :num_zeros] = 0
 
     return causal_mask
+
+
+def generate_token(
```

Comment:
NLDecoderEngine  --> calls generate_token(logits)

generate_token(logits) -> token
takes care of converting logits into token, i.e also handles all the sampling in between

sampling methods, I like if they are split into separate helper functions